### PR TITLE
Feature/162053589 post to non existing user

### PR DESCRIPTION
**Pull request by egonm12**

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -132,6 +131,18 @@ def active_for_authentication?
 
   private
 
+  def email_required?
```
Comment (ariejan) [inline app/models/user.rb]: Where are these used? You removed the email presence validation. 

`validates :email, presence: { unless: :virtual_user }` would also just work - no need for these helpers.

File: spec/graphql/mutations/post_mutations/create_post_spec.rb [hunk 2]
```diff
@@ -8,21 +8,54 @@
...
-        team_id: team.id,
-        kudos_meter_id: kudos_meter.id
-      }
-    }
     let(:ctx) { { current_user: users.first } }
 
-    it "creates a new post" do
+    it "creates a new post with existing users" do
+      args = {
+          message: Faker::Lorem.sentence(3),
```
Comment (ariejan) [inline spec/graphql/mutations/post_mutations/create_post_spec.rb]: If you don't need a message, don't stuff it full of data - it just takes up time and space unnecessarily. A simple `"message"` would suffice, unless you are testing parsing said message or something.

Comment (egonm12) [reply spec/graphql/mutations/post_mutations/create_post_spec.rb]:
```suggestion
          message: 'message',
```

File: spec/graphql/mutations/post_mutations/create_post_spec.rb [hunk 3]
```diff
@@ -8,21 +8,54 @@
...
+
+      expect do
+        subject.fields["createPost"].resolve(nil, args, ctx)
+      end.to change { Post.count }.by(1)
+     .and change { ActionMailer::Base.deliveries.count }.by(1)
+    end
+
+    it "creates a new post with non-existing users" do
+      args = {
+          message: Faker::Lorem.sentence(3),
```
Comment (ariejan) [inline spec/graphql/mutations/post_mutations/create_post_spec.rb]: Also change these?

File: spec/graphql/mutations/post_mutations/create_post_spec.rb [hunk 4]
```diff
@@ -8,21 +8,54 @@
...
+      expect do
+        subject.fields["createPost"].resolve(nil, args, ctx)
+      end.to change { Post.count }.by(1)
+      .and change { ActionMailer::Base.deliveries.count }.by(0)
+      .and change { User.count }.by(1)
+    end
+
+    it "creates a new post with existing and non-existing users" do
+      args = {
+          message: Faker::Lorem.sentence(3),
```
Comment (ariejan) [inline spec/graphql/mutations/post_mutations/create_post_spec.rb]: Also change these?

**Comment by ariejan** [comment]
Email validation is not broken, but already handled by Devise.

File: spec/graphql/mutations/post_mutations/create_post_spec.rb (hunk 4 above)
Comment (egonm12) [reply spec/graphql/mutations/post_mutations/create_post_spec.rb]:
```suggestion
          message: 'message',
```

File: spec/graphql/mutations/post_mutations/create_post_spec.rb (hunk 3 above)
Comment (egonm12) [reply spec/graphql/mutations/post_mutations/create_post_spec.rb]:
```suggestion
          message: 'message',
```

**Pull request merged**